clustering/methods.py

```diff
@@ -214,3 +216,70 @@ def generate_cluster_objects(
         ]
         clusters.append(Cluster(cluster_label, scooters))
     return clusters
+
+
+def compute_and_set_ideal_state(state: State, sample_size=None):
+    progressbar = Bar(
+        "| Computing ideal state", max=len(os.listdir(TEST_DATA_DIRECTORY))
+    )
+    number_of_scooters_counter = np.zeros(
+        (len(state.clusters), len(os.listdir(TEST_DATA_DIRECTORY)))
+    )
+    for index, file_path in enumerate(sorted(os.listdir(TEST_DATA_DIRECTORY))):
+        progressbar.next()
+        current_snapshot = read_bounded_csv_file(f"{TEST_DATA_DIRECTORY}/{file_path}")
+        if sample_size:
+            current_snapshot = current_snapshot.sample(sample_size)
+        current_snapshot["cluster"] = [
+            state.get_cluster_by_lat_lon(row["lat"], row["lon"]).id
+            for index, row in current_snapshot.iterrows()
+        ]
+        for cluster in state.clusters:
+            number_of_scooters_counter[cluster.id][index] = len(
+                current_snapshot[current_snapshot["cluster"] == cluster.id]
+            )
+    cluster_ideal_states = np.mean(number_of_scooters_counter, axis=1)
+    for cluster in state.clusters:
+        cluster.ideal_state = cluster_ideal_states[cluster.id]
+    progressbar.finish()
+
+
+def compute_and_set_trip_intensity(state: State, sample_size=None):
+    progress = Bar(
+        "| Computing trip intensity", max=len(os.listdir(TEST_DATA_DIRECTORY)),
+    )
+    # Fetch all snapshots from test data
+    trip_counter = np.zeros((len(state.clusters), len(os.listdir(TEST_DATA_DIRECTORY))))
```

Comment:
You could have made this one dimensional and saved memory space, but the solution is fine this time